pkg/controller/server_agent.go

```diff
@@ -30,8 +30,6 @@ func (s *server) apiAgentGetDelete(rw http.ResponseWriter, r *http.Request) {
 		s.apiAgentGet(rw, r, id)
 	case http.MethodDelete:
 		s.apiAgentDelete(rw, r, id)
```

Comment:
- This should be updated to use path parameter matching.
- Should remove this function and use the GET/DELETE route handler directly.